sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/FunctionToolCallingSample.java

```diff
@@ -0,0 +1,248 @@
+        // A list of tools to use (currently only has one tool type, function tool).
+        ChatCompletionsOptions chatCompletionsOptions = new ChatCompletionsOptions(chatRequestMessages)
+                .setTools(
+                        Arrays.asList(
+                                getCurrentWeatherToolFunction,
+                                getNDayWeatherForecastToolFunction
+                        )
+                );
+
+        // We can force the model to use a specific function, for example get_n_day_weather_forecast by using the
+        // function_call argument. By doing so, we force the model to make assumptions about how to use it.
+        chatCompletionsOptions.setToolChoice(BinaryData.fromObject(getNDayWeatherForecastToolFunction));
+
+        // We can also force the model to not use a function at all. By doing so we prevent it from producing a proper function calling.
+        // chatCompletionsOptions.setToolChoice(BinaryData.fromString("none"));
+
+        ChatCompletions chatCompletions = client.getChatCompletions(deploymentOrModelId, chatCompletionsOptions);
+
+        // Take your function tool calling result as the input prompt to make another request to service.
+        chatRequestMessages = handleFunctionToolResponse(chatCompletions.getChoices(), chatRequestMessages);
+        ChatCompletions chatCompletionsAnswer = client.getChatCompletions(deploymentOrModelId,
+                new ChatCompletionsOptions(chatRequestMessages));
+        getContentFromMessageResponse(chatCompletionsAnswer);
+    }
+
+    private static void parallelFunctionCallingSample(OpenAIClient client, String deploymentOrModelId) {
+        List<ChatRequestMessage> chatRequestMessages = new ArrayList<>();
+        chatRequestMessages.add(new ChatRequestSystemMessage("Don't make assumptions about what values to plug into functions. Ask for clarification if a user request is ambiguous."));
+        chatRequestMessages.add(new ChatRequestUserMessage("what is the weather going to be like in San Francisco and Glasgow over the next 4 days"));
+
+        ChatCompletionsOptions chatCompletionsOptions = new ChatCompletionsOptions(chatRequestMessages)
+                .setTools(
+                        Arrays.asList(
+                                getCurrentWeatherToolFunction,
+                                getNDayWeatherForecastToolFunction
+                        )
+                );
+        // Newer models like gpt-4-1106-preview or gpt-3.5-turbo-1106 can call multiple functions in one turn.
+        ChatCompletions chatCompletions = client.getChatCompletions(deploymentOrModelId, chatCompletionsOptions);
+        chatRequestMessages = handleFunctionToolResponse(chatCompletions.getChoices(), chatRequestMessages);
+
+        ChatCompletions chatCompletionsAnswer = client.getChatCompletions(deploymentOrModelId,
+                new ChatCompletionsOptions(chatRequestMessages));
+        getContentFromMessageResponse(chatCompletionsAnswer);
```

Comment:
I think that where this sample falls a bit short is in that it should also showcase what to do with the response, once the LLM requests the `tool_call` here. 

Then you could really showcase that you are calling things in parallel by doing something like:

```java
Flux.zip(
  Mono.fromCallable(() -> getCurrentWeatherToolFunction()),
  Mono.fromCallable(() -> getNDayWeatherForecastToolFunction()),
  Pair::new
).flatMap( pair -> {
  String first = pair.getFirst();
  String second = pair.getSecond();
  // ... the rest of the code
  client.getChatCompletions(...)
}).subscribe(...);
```

Reply:
Here I am pulling `Pair<T, K>` out of thin air. Here is a copilot implementation of that:

```java
    class Pair<T, K> {
        private T first;
        private K second;

        public Pair(T first, K second) {
            this.first = first;
            this.second = second;
        }

        public T getFirst() {
            return first;
        }

        public K getSecond() {
            return second;
        }
    }
```

Reply:
I am just using the names of the function as you have set them, but it seems a bit weird to call those functions like that. Maybe we could use different functions for this particular example.